### PR TITLE
Add kernel port implementation

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/forwarding/fwdport/ports/kernel.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	log "github.com/golang/glog"
+	fwdpb "github.com/openconfig/lemming/proto/forwarding"
+)
+
+func init() {
+	fwdport.Register(fwdpb.PortType_PORT_TYPE_KERNEL, kernelBuilder{})
+}
+
+type kernelBuilder struct {
+}
```
Comment (wenovus) [inline dataplane/forwarding/fwdport/ports/kernel.go:38]: a short comment please. I think my main confusion when starting to read this is what a "kernel" port is in the context of this code. From the proto it looks like it's a linux network device identified by a name. Would be nice to see a comment for this so it's clear from the outset.

File: dataplane/forwarding/fwdport/ports/kernel.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	if p.input, err = fwdaction.NewActions(kernelUpd.Kernel.GetInputs(), p.ctx); err != nil {
+		return fmt.Errorf("ports: input actions for port %v failed, err %v", p, err)
+	}
+	if p.output, err = fwdaction.NewActions(kernelUpd.Kernel.GetOutputs(), p.ctx); err != nil {
+		return fmt.Errorf("ports: output actions for port %v failed, err %v", p, err)
+	}
+	return nil
+}
+
+func (p *kernelPort) read() {
```
Comment (wenovus) [inline dataplane/forwarding/fwdport/ports/kernel.go:118]: Should this be named process() instead?

File: dataplane/forwarding/fwdport/ports/kernel.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	switch dir {
+	case fwdpb.PortAction_PORT_ACTION_INPUT:
+		return p.input
+	case fwdpb.PortAction_PORT_ACTION_OUTPUT:
+		return p.output
+	}
+	return nil
+}
+
+// State manages the state of the port.
```
Comment (wenovus) [inline dataplane/forwarding/fwdport/ports/kernel.go:160]: returns or manages? I guess it's not using `op` right now, so is there a TODO?

File: dataplane/forwarding/fwdport/ports/kernel.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+func (p *kernelPort) Cleanup() {
+	p.input.Cleanup()
+	p.output.Cleanup()
+	p.handle.Close()
+	p.input = nil
+	p.output = nil
+}
+
+// Update updates a port.
```
Comment (wenovus) [inline dataplane/forwarding/fwdport/ports/kernel.go:95]: updates what of a port?

File: dataplane/forwarding/fwdport/ports/kernel.go (hunk 3 above)
Comment (DanG100) [reply dataplane/forwarding/fwdport/ports/kernel.go]: added todo

File: dataplane/forwarding/fwdport/ports/kernel.go (hunk 2 above)
Comment (DanG100) [reply dataplane/forwarding/fwdport/ports/kernel.go]: sure

**Review by wenovus — APPROVED** [review]

**Pull request merged**